libparsers/tests/csv_rw_route/preprocessor.cpp

```diff
@@ -0,0 +1,139 @@
+#include "csv_rw_route.hpp"
+#include <doctest.h>
+#include <fstream>
+
+using namespace std::string_literals;
+namespace cs = parsers::csv_rw_route;
+namespace {
+std::string file_func(std::string const& base, std::string const& rel) {
+	return base + "/" + rel;
+}
+
+void write_to_file(std::string const& directive) {
+	std::ofstream file;
+	file.open("directive.csv", std::ios::trunc);
+	file.write(directive.c_str(), sizeof(char) * directive.size());
+}
+
+cs::preprocessed_lines setup(std::string const& test,
+                             parsers::errors::multi_error_t& output_errors) {
+	write_to_file(test);
+	int const seed = 1;
+	auto rng = openbve2::datatypes::rng{seed};
+	auto processed = cs::process_include_directives("directive.csv", rng, output_errors,
+	                                                cs::file_type::csv, file_func);
+	cs::preprocess_file(processed, rng, output_errors, cs::file_type::csv);
+	return processed;
+}
+
+} // namespace
+TEST_SUITE_BEGIN("libparser - csv_rw_route - preprocessor");
+
+TEST_CASE("libparsers - csv_rw_route - preprocessor - $chr") {
+	std::string test_command =
+	    "$Chr(75),$Chr(69),$Chr(86),$Chr(73),$Chr(78),\n$Chr(75),$Chr(69),$Chr(86),$Chr(73),$Chr(78)"s;
+	parsers::errors::multi_error_t output_errors;
+	auto processed = setup(test_command, output_errors);
+	CHECK_EQ(processed.lines.size(), 10);
+	std::string value{};
```

Comment:
You really need to reserve here, but I can't decide if it's worth the extra loop to get the size. Because these are tests, I am inclined not to give a fuck.